cmd/dlt/ocmrole/cmd.go

```diff
@@ -162,12 +161,12 @@ func run(cmd *cobra.Command, argv []string) error {
 		os.Exit(1)
 	}
 
-	roleExistOnAWS, _, err := r.AWSClient.CheckRoleExists(roleName)
+	roleExistOnAWS, existingRoleARN, err := r.AWSClient.CheckRoleExists(roleName)
 	if err != nil {
 		r.Reporter.Errorf("%v", err)
 	}
-	if !roleExistOnAWS {
-		r.Reporter.Warnf("ocm-role '%s' doesn't exist on the aws account %s", roleName, r.Creator.AccountID)
+	if !roleExistOnAWS || existingRoleARN != roleARN {
```

Comment:
the name is unique. but user can provide arn with path A while the existing role has arn with path B. We need to warn the user.

Reply:
Ok, I understand now the issue, thanks for the clarification.
Should we `exit` in this case and return an error? 

I think for the current flow we delete the role even if the role ARN is different.
This is a different case than `!roleExistOnAWS`.

Reply:
until the path change there was not difference in ARN. so if role existed it had to be same ARN. now situation is different. user is asking us to delete A and we see B. We need to report error.

Reply:
What I meant is that we should return an error if **existingRoleARN != roleARN** and quit the program, to avoid the next scenario:
I passed the wrong ARN (with path) and the role was deleted.
![image](https://user-images.githubusercontent.com/57869309/189519702-69cf49bf-c8fe-4f6e-a7f5-698d77e3184a.png)

It's a different scenario than `roleExistOnAWS` (where I deleted the role in the AWS console, but it's still linked):
![image](https://user-images.githubusercontent.com/57869309/189519866-17876720-e679-41d1-85d9-7759f87aa3ee.png)

Therefor I suggest to handle `!roleExistOnAWS` and `existingRoleARN != roleARN` differently.
WDYT?

Reply:
fixed